lib/web/mage/apply/main.js

```diff
@@ -49,6 +53,42 @@ define([
         });
     }
 
+    /**
+     * Initializes components assigned to a specified element via mage.applyFor
+     *
+     * @param {HTMLElement} el
+     * @param {Object|String} config
+     * @param {String} component
+     */
+    function applyFor(el, config, component) {
+        if (!el || isInViewport(el)) {
+            init(el, config, component)
+        } else {
+            document.addEventListener('mousemove', function (e) {
+                init(el, config, component)
+            }, { once: true });
+        }
+    }
+
+    /**
+     * Check if an element is visible in the viewport
+     *
+     * @param {HTMLElement} el
+     * @returns bool
+     */
+    function isInViewport(el) {
+        if (!el.checkVisibility()) return false;
```

Comment:
@rogerdz 
This one will throw errors once `el` will come as jQuery element which will be the case for some page builder widgets, I would add additional check like `if ((!_.isFunction(el.checkVisibility)) || !el.checkVisibility()) return false;`